src/System.Security.Cryptography.Cng/src/System/Security/Cryptography/ECCng.ImportExport.cs

```diff
@@ -0,0 +1,401 @@
+                }
+
+                return ImportKeyBlob(blob, parameters.Curve, includePrivateParameters);
+            }
+        }
+
+        internal static CngKey ImportPrimeCurveParameters(ECParameters parameters, bool includePrivateParameters)
+        {
+            Debug.Assert(parameters.Curve.IsPrime);
+
+            ECCurve curve = parameters.Curve;
+
+            unsafe
+            {
+                // We need to build a key blob structured as follows:
+                //     BCRYPT_ECCFULLKEY_BLOB       header
+                //     byte[cbFieldLength]          P
+                //     byte[cbFieldLength]          A
+                //     byte[cbFieldLength]          B
+                //     byte[cbFieldLength]          G.X
+                //     byte[cbFieldLength]          G.Y
+                //     byte[cbSubgroupOrder]        Order (n)
+                //     byte[cbCofactor]             Cofactor (h)
+                //     byte[cbSeed]                 Seed
+                //     byte[cbFieldLength]          Q.X
+                //     byte[cbFieldLength]          Q.Y
+                //     -- Only if "includePrivateParameters" is true --
+                //     byte[cbSubgroupOrder]        D
+
+                int blobSize = sizeof(BCRYPT_ECCFULLKEY_BLOB) +
+                    curve.Prime.Length +
+                    curve.A.Length +
+                    curve.B.Length +
+                    curve.G.X.Length +
+                    curve.G.Y.Length +
+                    curve.Order.Length +
+                    curve.Cofactor.Length +
+                    (curve.Seed == null ? 0 : curve.Seed.Length) +
+                    parameters.Q.X.Length +
+                    parameters.Q.Y.Length;
+
+                if (includePrivateParameters)
+                {
+                    blobSize += parameters.D.Length;
+                }
+
+                byte[] blob = new byte[blobSize];
+                fixed (byte* pBlob = blob)
+                {
+                    // Build the header
+                    BCRYPT_ECCFULLKEY_BLOB* pBcryptBlob = (BCRYPT_ECCFULLKEY_BLOB*)pBlob;
+                    pBcryptBlob->Version = 1; // No constant for this found in bcrypt.h
+                    pBcryptBlob->Magic = includePrivateParameters ?
+                        KeyBlobMagicNumber.BCRYPT_ECDSA_PRIVATE_GENERIC_MAGIC :
+                        KeyBlobMagicNumber.BCRYPT_ECDSA_PUBLIC_GENERIC_MAGIC;
+                    pBcryptBlob->cbCofactor = curve.Cofactor.Length;
+                    pBcryptBlob->cbFieldLength = parameters.Q.X.Length;
+                    pBcryptBlob->cbSeed = curve.Seed == null ? 0 : curve.Seed.Length;
+                    pBcryptBlob->cbSubgroupOrder = curve.Order.Length;
+                    pBcryptBlob->CurveGenerationAlgId = GetHashAlgorithmId(curve.Hash);
+                    pBcryptBlob->CurveType = (Interop.BCrypt.ECC_CURVE_TYPE_ENUM)curve.CurveType;
```

Comment:
I'd prefer if the enum to enum conversion was contained in a method.  Even if it is straight cast for now.  It's easier to find/fix if they become divergent (and the methods can assert that the value is one of the known ones)